src/Development/IDE/Core/Compile.hs

```diff
@@ -160,15 +164,13 @@ newtype RunSimplifier = RunSimplifier Bool
 compileModule
     :: RunSimplifier
     -> HscEnv
-    -> [(ModSummary, HomeModInfo)]
     -> TcModuleResult
     -> IO (IdeResult (SafeHaskellMode, CgGuts, ModDetails))
-compileModule (RunSimplifier simplify) packageState deps tmr =
+compileModule (RunSimplifier simplify) packageState tmr =
     fmap (either (, Nothing) (second Just)) $
     evalGhcEnv packageState $
         catchSrcErrors "compile" $ do
-            setupEnv (deps ++ [(tmrModSummary tmr, tmrModInfo tmr)])
-
+            setupEnv [(tmrModSummary tmr, tmrModInfo tmr)]
```

Comment:
surprised that we don't need to setup the env with the dependencies

Reply:
I've looked for every usage of this and made sure it is only called with `GhcSessionDeps`. In that case I don't think we need to call `setupEnv`